src/vscode-dts/vscode.proposed.chatOutputRenderer.d.ts

```diff
@@ -0,0 +1,61 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+declare module 'vscode' {
+
+	/**
+	 * Data returned from a tool.
+	 *
+	 * This is an opaque binary blob that can be rendered by a {@link ChatOutputRenderer}.
+	 */
+	export interface ToolResultDataOutput {
+		/**
+		 * The MIME type of the data.
+		 */
+		mime: string;
+
+		/**
+		 * The contents of the data.
+		 */
+		value: Uint8Array;
+	}
+
+	export interface ExtendedLanguageModelToolResult2 extends ExtendedLanguageModelToolResult {
+		// Temporary to allow `toolResultDetails` to return a ToolResultDataOutput
+		// TODO: Should we allow multiple per tool result?
+		toolResultDetails2?: Array<Uri | Location> | ToolResultDataOutput;
```

Comment:
It seems a little odd that we have an out-of-bounds way to represent data content in the same API that we can push data content to the chat response directly.

MCP solves with with an `audience: ('user' | 'assistant')[]` optional annotation on each of its content parts, which might be a more graceful way to represent this